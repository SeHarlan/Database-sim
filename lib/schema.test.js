const Schema = require('./schema.js');

const dogSchema = new Schema({
  name: {
    type: String,
    required: true
  },
  age: {
    type: Number,
    required: true
  },
  weight: {
    type: String
  }
});

const spot = {
  name: 'spot',
  age: 5,
  weight: '20 lbs'
};

const rover = {
  name: 'rover',
  age: '10'
};

const who = {
  age: 'hi'
};

describe('schema validation', () => {
  it('returns valid schemas', () => {
    const expectSpot = dogSchema.validate(spot);
    expect(expectSpot).toEqual({ name: 'spot', age: 5, weight: '20 lbs' });

    const expectRover = dogSchema.validate(rover);
    expect(expectRover).toEqual({ name: 'rover', age: 10, weight: null });

  });
  it('throws error for invalid input', () => {
    expect(() => dogSchema.validate(who)).toThrowErrorMatchingSnapshot();
  });
});