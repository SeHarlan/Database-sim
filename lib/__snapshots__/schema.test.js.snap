// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema validation throws error for invalid input 1`] = `"Missing required field >>name<<"`;
